src/adapter/src/coord/command_handler.rs

```diff
@@ -584,6 +584,12 @@ impl Coordinator {
             // `CREATE SOURCE` statements must be purified off the main
             // coordinator thread of control.
             Statement::CreateSource(stmt) => {
+                // Check privileges before purifying.
+                if let Err(e) =
+                    rbac::check_purify_create_source(&catalog, ctx.session(), &resolved_ids)
```

Comment:
This can't use `check_plan` because we don't have a plan yet?

Reply:
Yeah, exactly.